transformers/test_data/shared_behaviors/shared_flip_bid_trigger_suite.go

```diff
@@ -0,0 +1,407 @@
+				err := repo.Create(diffID, headerOne.Id, input.Metadata, newRepoValue)
+				Expect(err).NotTo(HaveOccurred())
+
+				var valueHistory []sql.NullString
+				queryErr := db.Select(&valueHistory, getFieldQuery)
+				Expect(queryErr).NotTo(HaveOccurred())
+				Expect(len(valueHistory)).To(Equal(2))
+				Expect(valueHistory[1].String).To(Equal(initialColumnVal))
+			})
+
+			It("ignores rows from earlier blocks", func() {
+				_, initialColumnVal := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				_, setupErr := db.Exec(insertFieldQuery, addressID, headerOne.BlockNumber, bidID, initialColumnVal, timestampOne)
+				Expect(setupErr).NotTo(HaveOccurred())
+
+				newRepoValue, _ := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				err := repo.Create(diffID, headerTwo.Id, input.Metadata, newRepoValue)
+				Expect(err).NotTo(HaveOccurred())
+
+				var valueHistory []sql.NullString
+				queryErr := db.Select(&valueHistory, getFieldQuery)
+				Expect(queryErr).NotTo(HaveOccurred())
+				Expect(len(valueHistory)).To(Equal(2))
+				Expect(valueHistory[0].String).To(Equal(initialColumnVal))
+			})
+
+			It("ignores rows from different address", func() {
+				_, initialColumnVal := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				differentAddressID, addressErr := shared.GetOrCreateAddress(test_data.RandomString(40), db)
+				Expect(addressErr).NotTo(HaveOccurred())
+				_, setupErr := db.Exec(insertFieldQuery, differentAddressID, headerTwo.BlockNumber, bidID, initialColumnVal, timestampTwo)
+				Expect(setupErr).NotTo(HaveOccurred())
+
+				newRepoValue, _ := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				err := repo.Create(diffID, headerOne.Id, input.Metadata, newRepoValue)
+				Expect(err).NotTo(HaveOccurred())
+
+				var valueHistory []sql.NullString
+				queryErr := db.Select(&valueHistory, getFieldQuery)
+				Expect(queryErr).NotTo(HaveOccurred())
+				Expect(len(valueHistory)).To(Equal(2))
+				Expect(valueHistory[1].String).To(Equal(initialColumnVal))
+			})
+
+			It("ignores rows with different bidID", func() {
+				_, initialColumnVal := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				differentBidID := rand.Int()
+				_, setupErr := db.Exec(insertFieldQuery, addressID, headerTwo.BlockNumber, differentBidID, initialColumnVal, timestampTwo)
+				Expect(setupErr).NotTo(HaveOccurred())
+
+				newRepoValue, _ := randomBidStorageValue(input.Metadata.Type, input.PackedValueType)
+				err := repo.Create(diffID, headerOne.Id, input.Metadata, newRepoValue)
+				Expect(err).NotTo(HaveOccurred())
+
+				var valueHistory []sql.NullString
+				queryErr := db.Select(&valueHistory, getFieldQuery)
+				Expect(queryErr).NotTo(HaveOccurred())
+				Expect(len(valueHistory)).To(Equal(2))
+				Expect(valueHistory[1].String).To(Equal(initialColumnVal))
+			})
+		})
+	})
+}
+
+func FlipBidSnapshotTriggerTests(input BidTriggerTestInput) {
```

Comment:
I had some trouble deciding how to balance abstraction vs readability with these tests. I ended up allowing some duplication for these 2 tests (the `inserting a new field` suite is duplicated on line 236) since they have to make assertions about the entire record, and thus are dealing with 2 distinct types for `flip` vs `flap/flop` records. Whereas the `updating history with a new value` suite only deals with 1 field at a time.

So, while it would've been _possible_ to abstract the suites on lines 163 and 263 into a single suite, I felt like it would be too detrimental to the tests' (already questionable) clarity.